haskell-src/Concordium/Types/ProtocolVersion.hs

```diff
@@ -140,162 +326,74 @@ promoteProtocolVersion P4 = SomeProtocolVersion SP4
--- * 'TOV1' is used in PV5 and onwards. The hash is computed via a merkle tree and the
--- exact reject reasons for failed transactions are omitted from the hash.
-data TransactionOutcomesVersion
-    = TOV0
-    | TOV1
-
--- |Projection of 'ProtocolVersion' to 'TransactionOutcomesVersion'.
-type family TransactionOutcomesVersionFor (pv :: ProtocolVersion) :: TransactionOutcomesVersion where
-    TransactionOutcomesVersionFor 'P1 = 'TOV0
-    TransactionOutcomesVersionFor 'P2 = 'TOV0
-    TransactionOutcomesVersionFor 'P3 = 'TOV0
-    TransactionOutcomesVersionFor 'P4 = 'TOV0
-    TransactionOutcomesVersionFor 'P5 = 'TOV1
-    TransactionOutcomesVersionFor 'P6 = 'TOV1
-
--- |Supporting type for bringing the 'TransactionOutcomesVersion' to the term level.
-data STransactionOutcomesVersion (tov :: TransactionOutcomesVersion) where
-    STOV0 :: STransactionOutcomesVersion 'TOV0
-    STOV1 :: STransactionOutcomesVersion 'TOV1
-
-class IsTransactionOutcomesVersion (tov :: TransactionOutcomesVersion) where
-    -- |The singleton associated with the outcomes version.
-    transactionOutcomesVersion :: STransactionOutcomesVersion tov
-
-instance IsTransactionOutcomesVersion 'TOV0 where
-    transactionOutcomesVersion = STOV0
-
-instance IsTransactionOutcomesVersion 'TOV1 where
-    transactionOutcomesVersion = STOV1
-
--- |A type used at the kind level to denote that delegation is or is not expected to be supported
--- at an account version. This is intended to give more descriptive type errors in cases where the
--- typechecker simplifies 'AVSupportsDelegationB'. In particular, a required constraint of
--- @AVSupportsDelegation 'AccountV0@ will give a type error:
---
--- @
---   Couldn't match type: 'DelegationNotSupported 'AccountV0
---   with: 'DelegationSupported 'AccountV0
--- @
+-- |Demote an 'SProtocolVersion' to a 'ProtocolVersion'.
+demoteProtocolVersion :: SProtocolVersion pv -> ProtocolVersion
+demoteProtocolVersion = fromSing
+
+-- |Constraint on a type level 'AccountVersion' that can be used to get a corresponding
+-- 'SAccountVersion' (see 'accountVersion'). (An alias for 'SingI'.)
+type IsAccountVersion (av :: AccountVersion) = SingI av
+
+-- |Constraint on a type level 'ChainParametersVersion' that can be used to get a corresponding
+-- 'SChainParametersVersion' (see 'chainParametersVersion'). (An alias for 'SingI'.)
+type IsChainParametersVersion (cpv :: ChainParametersVersion) = SingI cpv
+
+-- |Constraint on a type level 'TransactionOutcomesVersion' that can be used to get a corresponding
+-- 'STransactionOutcomesVersion' (see 'transactionOutcomesVersion'). (An alias for 'SingI'.)
+type IsTransactionOutcomesVersion (tov :: TransactionOutcomesVersion) = SingI tov
+
+-- |Constraint on a type level 'TransactionOutcomesVersion' that can be used to get a corresponding
+-- 'STransactionOutcomesVersion' (see 'transactionOutcomesVersion'). This wraps 'SingI', but also
```

Comment:
Good catch, I think in the first part of the doc it should be `ProtocolVersion` instead of `TransactionOutcomesVersion` in order to make sense. At least then the sentence parses .. :) 

Fixed.